src/RenderManager.cpp

```diff
@@ -64,19 +56,18 @@ void RenderManager::AddTileset(BYTE* tilesetMemoryPointer, std::size_t tilesetSi
 
 void RenderManager::AddTileset(std::string filename, ImageFormat imageFormat)
 {
-	FIBITMAP* fiTilesetBmp = FreeImage_Load(GetFiImageFormat(imageFormat), filename.c_str());
+	FreeImageBmp freeImageBmp(GetFiImageFormat(imageFormat), filename.c_str());
 
-	ScaleTileset(fiTilesetBmp);
+	ScaleTileset(freeImageBmp);
 }
 
-void RenderManager::ScaleTileset(FIBITMAP* fiTilesetBmp)
+void RenderManager::ScaleTileset(FreeImageBmp& fiTilesetBmp)
 {
 	const unsigned nonScaledTileLength = 32;
-	const unsigned tilesetScaledWidth = FreeImage_GetWidth(fiTilesetBmp) / nonScaledTileLength * scaleFactor;
-	const unsigned tilesetScaledHeight = FreeImage_GetHeight(fiTilesetBmp) / nonScaledTileLength * scaleFactor;
+	const unsigned tilesetScaledWidth = FreeImage_GetWidth(fiTilesetBmp.fiBitmap) / nonScaledTileLength * scaleFactor;
+	const unsigned tilesetScaledHeight = FreeImage_GetHeight(fiTilesetBmp.fiBitmap) / nonScaledTileLength * scaleFactor;
```

Comment:
I suppose you could add convenience methods on `FreeImageBmp`. Ex:
```cpp
const unsigned tilesetScaledWidth = fiTilesetBmp.GetWidth() / nonScaledTileLength * scaleFactor;
```

That might also reduce the pressure to provide direct raw access to the `.fiBitmap` field. Granted, the field is const, and allowing access may be good for ease of extensibility and library upgrades.

Reply:
My plan was to just to cover disposal of the bitmap. Hadn't really thought about fully hiding FIBITMAP in the class FreeImageBmp. Not that I'm opposed to the idea.